Documentation/Evolution/CharacterClasses.md

```diff
@@ -0,0 +1,354 @@
+
+```swift
+Pattern {
+    "abc"
+}
+```
+
+### Match any: `.`, `\X`
+
+The dot metacharacter matches any single character or element except for a newline (see the _Whitespace and Newlines_ section, below). In "dot matches all" mode, newlines are treated like any other character. A future proposal will discuss modes in more detail.
+
+`\X` matches any grapheme cluster (`Character`), even when the regular expression is otherwise matching at the Unicode scalar or POSIX level.
+
+**DSL:** The "match anything" pattern is spelled `.anyCharacter`. `.anyCharacter` will match at the level of the overall pattern unless given a more specific level, so `.anyCharacter.graphemeScalarSemantics` is equivalent to the `\X` metacharacter.
+
+```swift
+Pattern {
+    OneOrMore(.anyCharacter)
+}
+```
+
+### Digits: `\d`,`\D`
+
+`\d` matches a "digit", which is any `Character` or Unicode scalar with the Unicode general category `Decimal_Number`.
+
+| Matching Level | Note |
+|-------------------------------|---------|
+| Character | any character where the general category of the first unicode scalar is `Decimal_Number` |
+| Unicode Scalar | any scalar where the general category is `Decimal_Number`, not including any combining marks |
+| POSIX | `[0-9]` |
+
+`\D` matches the inverse set of characters/elements.
+
+**DSL:** The corresponding pattern is called `.digitCharacter`:
+
+```swift
+Pattern {
+    OneOrMore(.digitCharacter)
+}
+```
+
+
+### Word Characters: `\w`, `\W`
+
+`\w` matches any letter or decimal number, as indicated by the Unicode general categories:
+
+- `Uppercase_Letter`
+- `Lowercase_Letter`
+- `Titlecase_Letter`
+- `Other_Letter`
+- `Decimal_Number`
+
+| Matching Level | Note |
+|-------------------------------|---------|
+| Character | any character where the general category of the first unicode scalar is in the above list |
+| Unicode Scalar | any scalar where the general category is in the above list |
+| POSIX | `[A-Za-z0-9_]` |
+
+`\W` matches the inverse set of characters/elements.
+
+**DSL:** The corresponding pattern is called `.wordCharacter`:
+
+```swift
+Pattern {
+    OneOrMore(.wordCharacter)
+}
+```
+
+
+### Whitespace and Newlines: `\s`, `\S` (plus `\h`, `\H`, `\v`, `\V`, and `\R`)
+
+`\s` matches any single, non-zero width whitespace character, as denoted by the following list.
+
+- `CHARACTER TABULATION` (U+0009)
+- `LINE FEED (LF)` (U+000A)*
+- `LINE TABULATION` (U+000B)*
+- `FORM FEED (FF)` (U+000C)*
+- `CARRIAGE RETURN (CR)` (U+000D)*
+- `NEWLINE (NEL)` (U+0085)*
+- any character in the Unicode general category `Z`/`Separator`
+- the CR/LF sequence
+
+`\h` matches the tab character (U+0009) or any character in the Unicode general category `Space_Separator`.
+
+`\v` matches the five line separator characters above, marked with *, as well as (U+2028) and (U+2029), not including the CR/LF sequence.
+
+`\R` matches the same characters as `\v`, and in addition matches the CR/LF sequence.
+
```

Comment:
What about `Character.isWhitespace`? What about `Unicode.Scalar.Properties.isWhitespace`? How does this fit together?